src/app/events/tasks.ts

```diff
@@ -968,6 +1059,48 @@ ${this.getLink()}
         }
       }
 
+    const cleanupCb = (success: boolean) => {
+      if (success) saved?.()
+      if (doLocks) Locks.unlock(this.id)
+      if (assign) this.assignToMe()
+    }
+
+    if (isDrive) {
+      await this.openDriverEditDialog(ui, cleanupCb)
+    } else {
+      await this.openConsumerEditDialog(ui, cleanupCb)
+    }
+  }
+
+  private async openDriverEditDialog(ui: UITools, cleanupCb: (ok: boolean) => void) {
+    const e = this.$
+    ui.areaDialog({
+      title: 'פרטי נסיעה',
+      fields: [
+        [e.category!],
+        e.title,
+        e.spaceAvailable,
+        e.description,
+        e.address,
+        e.toAddress,
+        e.privateDriverNotes,
+        [e.eventDate, e.startTime, e.relevantHours],
+        e.imageId,
+        e.internalComments,
+        e.externalId,
+      ],
+      ok: () => {
+        this.save().then(() => cleanupCb(true)).then(() => { this.save(); })
```

Comment:
You're missing an await here - the causes an error where users forget to fill out something it closes the dialog and shows them the error without allowing them a chance to get back to it

Reply:
![image](https://github.com/noam-honig/trempim/assets/16635859/ddd0b3a7-97b6-4e8f-b921-aa5a2cd9c0cc)